code/campaignclassic/src/main/java/com/adobe/marketing/mobile/AEPPushTemplateBroadcastReceiver.java

```diff
@@ -0,0 +1,302 @@
+/*
+  Copyright 2023 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+package com.adobe.marketing.mobile;
+
+import android.app.Notification;
+import android.app.PendingIntent;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.os.Bundle;
+import android.widget.RemoteViews;
+import androidx.core.app.NotificationCompat;
+import androidx.core.app.NotificationManagerCompat;
+import com.adobe.marketing.mobile.campaignclassic.R;
+import com.adobe.marketing.mobile.services.ServiceProvider;
+import com.adobe.marketing.mobile.services.caching.CacheResult;
+import com.adobe.marketing.mobile.services.caching.CacheService;
+import com.adobe.marketing.mobile.util.StringUtils;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+public class AEPPushTemplateBroadcastReceiver extends BroadcastReceiver {
+
+    @Override
+    public void onReceive(final Context context, final Intent intent) {
+        if (intent.getAction().equals(CampaignPushConstants.IntentActions.FILMSTRIP_LEFT_CLICKED)
+                || intent.getAction()
+                        .equals(CampaignPushConstants.IntentActions.FILMSTRIP_RIGHT_CLICKED)) {
+            handleFilmstripNotificationIntent(context, intent);
+        }
+    }
+
+    private void handleFilmstripNotificationIntent(final Context context, final Intent intent) {
```

Comment:
What are your thoughts on moving this into the `CarouselTemplateNotificationBuilder`? That class already knows how to build a notification from a `RemoteMessage` payload. Now we can also have it build a notification from an intent.

Reply:
this makes sense, i'll move it